x-pack/legacy/plugins/ml/public/jobs/new_job_new/common/job_creator/util/general.ts

```diff
@@ -208,3 +212,25 @@ export function advancedStartDatafeed(jobCreator: JobCreatorType) {
 export function aggFieldPairsCanBeCharted(afs: AggFieldPair[]) {
   return afs.some(a => a.agg.dslName === null) === false;
 }
+
+export async function autoSetJobCreatorTimeRange(
```

Comment:
I added a small function `getTimeRange` in `x-pack/legacy/plugins/ml/public/jobs/new_job_new/recognize/page.tsx` which does almost the same. Maybe it'd convenient to have `JobCreatorService` which contaians common methods?

Reply:
this function has been moved into the advanced job creator class as it's more suited there